pkg/cluster/context.go

```diff
@@ -158,7 +158,7 @@ func (c *Context) Create(cfg *config.Config, retain bool, wait time.Duration) er
 	// Kubernetes cluster; please note that the list of actions automatically
 	// adapt to the topology defined in config
 	// TODO(fabrizio pandini): make the list of executed actions configurable from CLI
-	err = cc.Exec(nodeList, []string{"config", "init", "join"}, wait)
+	err = cc.Exec(nodeList, []string{"haproxy", "config", "init", "join"}, wait)
```

Comment:
BTW I still intend to refactor this some soon, thinking about how we can make it harder to do wrong (EG strings that aren't valid), and more highly concurrent for those speedy test times :upside_down_face: 

Making it internal for now allows us to delay that in favor of landing awesome stuff like this PR :D

Reply:
Concurrency will give a boost on the multi-node use cases, while for single node I assume this will be not relevant. In turn, we will lose something on the UX (e.g the nice spinner or the readability of the command output)

I think that as of today it is already possible to introduce concurrency on the node provisioning part; actions instead have an intrinsic notion of order that should be considered, but it is possible to get some improvements there as well. Whenever you want to do this ping me and I will share a possible solution  😉.

I got your point about keeping things internal, but there are also some backside in this as well; this makes this point an interesting topic the upcoming kind meetings!

Reply:
yeah totally agree with those, been thinking about this one too. the UX is tough but I do think it is important to keep the time minimal, fast iteration loops make developers (like me!) happy and I think we can find a good middle ground. 

again have a few half baked ideas to discuss soon, but I don't want to introduce the complexity right now either :-)